### PR TITLE
EOS-22089: MessageBus logger not initialized

### PR DESCRIPTION
**Pull request by suryakumar1024**
Signed-off-by: suryakumar.kumaravelan <suryakumar.kumaravelan@seagate.com>

# Problem Statement
- MessageBus logger not initialized on error cases
```
File "/usr/lib/python3.6/site-packages/cortx/utils/message_bus/message_bus.py", 
line 45, in __init__#012    
Log.error(f"MessageBusError: {e.rc} Error while parsing" \#012  
File "/usr/lib/python3.6/site-packages/cortx/utils/log.py", 
line 118, in error#012    
Log.logger.error(f"[{caller}] {msg}", *args, **kwargs)#012
AttributeError: 'NoneType' object has no attribute 'error'

```

# Design
-  For Bug describe the fix here. 
-  For Feature, Post the link to the solution page on the confluence CORTX Foundation Library 

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: 
-  Confirm All CODACY errors are resolved [Y/N]: 

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [ ] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] Jira is updated
- [x] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/message_bus/message_bus.py [hunk 1]
```diff
@@ -32,14 +32,14 @@ class MessageBus(metaclass=Singleton):
     def __init__(self):
         """ Initialize a MessageBus and load its configurations """
         try:
-            Conf.load('message_bus', self.conf_file)
             Conf.load('config_file', 'json:///etc/cortx/cortx.conf')
             log_level = Conf.get('config_file', 'utils>log_level', 'INFO')
+            Log.init("MessageBus", '/var/log/cortx/utils/message_bus',
+                level=log_level, backup_count=5, file_size_in_mb=5)
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/message_bus/message_bus.py:38]: @suryakumar1024 
Please move Log.init() outside the try clause.

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/message_bus/message_bus.py (hunk 1 above)
Comment (suryakumar1024) [reply py-utils/src/utils/message_bus/message_bus.py]: Updated in recent patch

**Review by selvakumaar5496 — CHANGES_REQUESTED** [review]
There are multiple spaces in the follow-up lines. Please correct them in all lines.
We have agreed to have only 4 spaces indent.

File: py-utils/src/utils/message_bus/message_bus.py [hunk 2]
```diff
@@ -31,29 +31,29 @@ class MessageBus(metaclass=Singleton):
 
     def __init__(self):
         """ Initialize a MessageBus and load its configurations """
+        Conf.load('config_file', 'json:///etc/cortx/cortx.conf')
+        log_level = Conf.get('config_file', 'utils>log_level', 'INFO')
+        Log.init("MessageBus", '/var/log/cortx/utils/message_bus',
```
Comment (ujjwalpl) [inline py-utils/src/utils/message_bus/message_bus.py:36]: Why MessageBus ? We dont use canonical naming anywhere in the code except for class names

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/message_bus/message_bus.py [hunk 3]
```diff
@@ -31,29 +31,30 @@ class MessageBus(metaclass=Singleton):
...
-                level=log_level, backup_count=5, file_size_in_mb=5)
             Log.info(f"MessageBus initialized as {broker_type}")
         except ConfError as e:
             Log.error(f"MessageBusError: {e.rc} Error while parsing" \
-                      f" configuration file {self.conf_file}. {e}.")
-            raise MessageBusError(e.rc, "Error while parsing " +\
-                "configuration file %s. %s.", self.conf_file, e)
+                f" configuration file {self.conf_file}. {e}.")
+            raise MessageBusError(e.rc, "Error while parsing " + \
+                                  "configuration file %s. %s.", self.conf_file, e)
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/message_bus/message_bus.py:49]: Please follow the style guide for line-break indentations

File: py-utils/src/utils/message_bus/message_bus.py [hunk 4]
```diff
@@ -31,29 +31,30 @@ class MessageBus(metaclass=Singleton):
...
             Log.error(f"MessageBusError: {e.rc} Error while parsing" \
-                      f" configuration file {self.conf_file}. {e}.")
-            raise MessageBusError(errno.ENOENT, "Error while parsing " +\
+                f" configuration file {self.conf_file}. {e}.")
+            raise MessageBusError(errno.ENOENT, "Error while parsing " + \
                 "configuration file %s. %s.", self.conf_file, e)
 
         self._broker = MessageBrokerFactory.get_instance(broker_type, \
-                self._broker_conf)
+                                                         self._broker_conf)
```
Comment (rahul-tripathi-git) [inline py-utils/src/utils/message_bus/message_bus.py:57]: Please follow the style guide for line-break indentations

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/message_bus/message_bus.py (hunk 3 above)
Comment (suryakumar1024) [reply py-utils/src/utils/message_bus/message_bus.py]: Updated in recent patch.

File: py-utils/src/utils/message_bus/message_bus.py (hunk 4 above)
Comment (suryakumar1024) [reply py-utils/src/utils/message_bus/message_bus.py]: Updated in recent patch.

File: py-utils/src/utils/message_bus/message_bus.py (hunk 2 above)
Comment (suryakumar1024) [reply py-utils/src/utils/message_bus/message_bus.py]: Updated in recent patch.

**Review by rahul-tripathi-git — APPROVED** [review]
Looks good to me.

File: py-utils/src/utils/message_bus/message_bus.py [hunk 5]
```diff
@@ -31,29 +31,30 @@ class MessageBus(metaclass=Singleton):
...
     def __init__(self):
         """ Initialize a MessageBus and load its configurations """
-        try:
-            Conf.load('message_bus', self.conf_file)
-            Conf.load('config_file', 'json:///etc/cortx/cortx.conf')
-            log_level = Conf.get('config_file', 'utils>log_level', 'INFO')
+        Conf.load('config_file', 'json:///etc/cortx/cortx.conf',
+            skip_reload=True)
+        log_level = Conf.get('config_file', 'utils>log_level', 'INFO')
+        Log.init("message_bus", '/var/log/cortx/utils/message_bus',
```
Comment (vastradparayya) [inline py-utils/src/utils/message_bus/message_bus.py:37]: @suryakumar1024 Should be single quotes as single word.
"message_bus" -> 'message_bus'

**Review by sachinpunadikar — APPROVED** [review]
+1

File: py-utils/src/utils/message_bus/message_bus.py (hunk 5 above)
Comment (suryakumar1024) [reply py-utils/src/utils/message_bus/message_bus.py]: Updated in recent patch

**Review by vastradparayya — APPROVED** [review]
Looks good to me

**Review by ujjwalpl — APPROVED** [review]

**Pull request merged**